eva-tools/src/main/java/embl/ebi/variation/eva/vcfdump/BiodataVariantToVariantContextConverter.java

```diff
@@ -0,0 +1,183 @@
+                try {
+                    regionSequence = cellbaseClient.getSequence(new Region(variant.getChromosome(), regionStart, regionEnd));
+                } catch (Exception e) {
+                    throw new CellbaseSequenceDownloadError("Error getting from Cellbase sequence for Region " + variant.getChromosome() +
+                            ":" + regionStart + "-" + regionEnd, e);
+                }
+            }
+            String nucleotide = getNucleotideFromRegionSequence(start, region.getStart(), regionSequence);
+            return nucleotide;
+        } else {
+            throw new IllegalArgumentException(String.format(
+                    "CellBase was not provided, needed to fill empty alleles in variant %s:%d:%s>%s",
+                    variant.getChromosome(), variant.getStart(), variant.getReference(), variant.getAlternate()));
+        }
+    }
+
+    private String getNucleotideFromRegionSequence(int start, int regionStart, String regionSequence) {
+        int relativePosition = start - regionStart;
+        return regionSequence.substring(relativePosition, relativePosition + 1);
+    }
+
+    private Set<Genotype> getGenotypes(Variant variant, String[] allelesArray) {
+        Set<Genotype> genotypes = new HashSet<>();
+
+        Allele[] variantAlleles =
+                {Allele.create(allelesArray[0], true), Allele.create(allelesArray[1]), Allele.create(Allele.NO_CALL, false)};
+
+        for (VariantSource source : sources) {
+            List<VariantSourceEntry> variantStudyEntries =
+                    variant.getSourceEntries().values().stream().filter(s -> s.getStudyId().equals(source.getStudyId())).collect(Collectors.toList());
+            for (VariantSourceEntry variantStudyEntry : variantStudyEntries) {
+                genotypes = getStudyGenotypes(genotypes, variantAlleles, variantStudyEntry);
+            }
+        }
+        return genotypes;
+    }
+
+    private Set<Genotype> getStudyGenotypes(Set<Genotype> genotypes, Allele[] variantAlleles, VariantSourceEntry variantStudyEntry) {
+        for (Map.Entry<String, Map<String, String>> sampleEntry : variantStudyEntry.getSamplesData().entrySet()) {
+            String gt = sampleEntry.getValue().get(GENOTYPE_KEY);
+            org.opencb.biodata.models.feature.Genotype genotype = new org.opencb.biodata.models.feature.Genotype(gt, variantAlleles[0].getBaseString(), variantAlleles[1].getBaseString());
+            List<Allele> genotypeAlleles = new ArrayList<>(2);
+            for (int index : genotype.getAllelesIdx()) {
+                if (index == -1 || index > NO_CALL_ALLELE_INDEX) {
+                    index = NO_CALL_ALLELE_INDEX;
+                }
+                genotypeAlleles.add(variantAlleles[index]);
+            }
+
+            genotypes.add(new GenotypeBuilder().name(getFixedSampleName(variantStudyEntry.getStudyId(), sampleEntry.getKey())).alleles(genotypeAlleles).phased(genotype.isPhased()).make());
```

Comment:
I think I misunderstood this code: I thought it was setting phasing always to true (don't ask me why), but now I look into again it seems to behave correctly. I didn't mean not to use phased phenotypes anyway: projects such as 1000 Genomes have well defined them and we shouldn't discard that information.